src/core/shape_inference/include/inverse_shape_inference.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "dimension_util.hpp"
+#include "openvino/op/inverse.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v14 {
+template <class TShape, class TRShape = result_shape_t<TShape>>
+std::vector<TRShape> shape_infer(const Inverse* op, const std::vector<TShape>& input_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1);
+
+    const auto& input_shape = input_shapes[0];
+    const auto input_rank = input_shape.rank();
+    if (input_rank.is_static()) {
+        const auto size = input_shape.size();
+        NODE_VALIDATION_CHECK(op, size >= 2, "Input must be at least a 2D matrix.");
+
+        if (input_shape[size - 2].is_static() && input_shape[size - 1].is_static()) {
+            NODE_SHAPE_INFER_CHECK(op,
+                                   input_shapes,
+                                   input_shape[size - 2].get_min_length() == input_shape[size - 1].get_min_length(),
+                                   "Input must contain square matrices of the same shape.");
+        }
```

Comment:
```suggestion
        NODE_SHAPE_INFER_CHECK(op,
                               input_shapes,
                               input_shape[size - 2].compatible(input_shape[size - 1]),
                               "Input must contain square matrices of the same shape.");
```

This will check for dynamic shapes if they have common range dimensions like `[2,5]` and `[8,10]` will be rejected because when static they will never have same length.

Consider expand test to check such interval dimensions.

Reply:
Applied!